renderer_cocoa/lib/React_Components.re

```diff
@@ -0,0 +1,83 @@
+open React;
+open Cocoa;
+
+type color = {
+  red: float,
+  green: float,
+  blue: float,
+  alpha: float,
+};
+type style = {
+  backgroundColor: color,
+  borderWidth: float,
+  borderColor: color,
+};
+
+module View = {
+  let component = statelessNativeComponent("NSView");
+  let make = (~layout, ~style, children) => {
+    ...component,
+    render: _ => {
+      make: () => {
+        let view = NSView.make();
+        {view, layoutNode: makeLayoutNode(~layout, view)};
+      },
+      shouldReconfigureInstance: (~oldState as _, ~newState as _) => true,
+      updateInstance: (_self, {view}) => {
+        let {red, green, blue, alpha} = style.backgroundColor;
+        NSView.setBackgroundColor(view, red, green, blue, alpha);
+        NSView.setBorderWidth(view, style.borderWidth);
+        let {red, green, blue, alpha} = style.borderColor;
+        NSView.setBorderColor(view, red, green, blue, alpha);
+      },
+      children,
+    },
+  };
+  let createElement = (~layout, ~style, ~children, ()) =>
+    element(make(~layout, ~style, listToElement(children)));
+};
+
+module Button = {
+  let component = statelessNativeComponent("Button");
+  let make = (~title=?, ~layout, ~style=?, ~callback=() => (), children) => {
+    ...component,
+    render: _ => {
+      make: () => {
+        let btn = NSButton.make();
+        let btn =
+          switch (title) {
+          | Some(title) => NSButton.setTitle(btn, title)
+          | None => btn
+          };
+        let btn =
+          NSButton.setCallback(
+            btn,
+            () => {
+              callback();
+              RunLoop.loop();
+            },
+          );
+
+        {view: btn, layoutNode: makeLayoutNode(~layout, btn)};
+      },
+      shouldReconfigureInstance: (~oldState as _, ~newState as _) => true,
+      updateInstance: (_self, {view}) =>
+        switch (style) {
+        | Some(style) =>
+          let {red, green, blue, alpha} = style.backgroundColor;
+
+          NSView.setBackgroundColor(view, red, green, blue, alpha);
+          NSView.setBorderWidth(view, style.borderWidth);
+          let {red, green, blue, alpha} = style.borderColor;
+          NSView.setBorderColor(view, red, green, blue, alpha);
```

Comment:
We should probably make one call over the bridge with the style when we figure out what it is.

Reply:
Hmm, the only reason why I did this is to avoid messing with 'stubs' side. Why bother if we're going to remove alloc calls later?

Reply:
yea, that's a fair point.